Add Faker::Games::DungeonsAndDragons

jameslvdb opened the pull request:
Created `Faker::Games::DungeonsAndDragons` generator. It currently supports the following methods:
- `ability`: returns one of the six standard D&D abilities.
- `skill`: returns a random skill. If passed one of the abilities from above, it will return a skill corresponding to that ability.
- `alignment`: returns one of the nine D&D alignments.
- `armor`: returns the name of a piece of armor.
- `player_class`: returns the name of one of D&D's core classes (used `player_class` to avoid conflict with the `class` reserved word
- `subclass`: returns a random subclass. If passed one of the classes from above, it will return a subclass corresponding to that class.
- `language`: returns the name of one of the languages from the D&D universe.
- `monster`: returns the name of a monster from D&D.
- `race`: returns the name of a race from D&D.
- `subrace`: returns a random subrace. If passed one of the races from above, it will return a corresponding subrace.
- `spell`: returns the name of a spell.
- `magic_school`: returns the name of one of the schools of magic.
- `weapon`: returns the name of a weapon.
- `magic_item` returns the name of a magic item.

This is a large generator (41K) -- I went a little all-out. If any maintainers would like this generator to be smaller before approving this PR, I'll be happy to make some updates. Even if the size is okay, I'm a little on the fence about keeping the `language`, `weapon`, and `armor` methods around. While these are important to D&D, they are likely served just fine by other generators for the purposes of testing, and they're small enough collections that if one does feel strongly about having an accurate generator, adding them locally on a per-project basis wouldn't be difficult.

Zeragamba commented:
Oh, hold up. We failed our perception checks... [`Faker::Games:DnD`](https://github.com/faker-ruby/faker/blob/master/lib/faker/games/dnd.rb) already exists. Can we merge all of this into there?

jameslvdb commented:
@Zeragamba I think there would be some duplication here if we merge as is. The previous PR refers to races as 'species', while this one uses 'races' (which reflects the terminology in the Player's Handbook). I'd be happy to pull down again and make updates to make it all line up!

Zeragamba commented:
Merge faker/master into your local branch and you should be good to go :)

dollerbill commented:
@jameslvdb I did a little bit of cleanup with #2098 which might make a refactor of this PR a bit easier. I reintroduced `races` as well as added `language` and `monster`. I added weapons but broke them down between `melee` and `ranged`. It looks like everything else covered with your additions shouldn't have any conflicts, so could be a pretty easy merge

jameslvdb commented:
Thanks @dollerbill! Honestly forgot about this PR, thanks for reminding me 😆 

Since the `DnD` class has already been introduced, I went ahead and moved all of my additions into `DnD` and removed all of the `DungeonsAndDragons`-related code. 

I also added `.tool-versions` to the `.gitignore` file for the `asdf` users who might contribute in the future 😄 

@Zeragamba is this what you were looking for?

dollerbill commented on README.md:
```diff
@@ -235,6 +235,7 @@ gem 'faker', :git => 'https://github.com/faker-ruby/faker.git', :branch => 'mast
   - [Faker::Game](doc/games/game.md)
   - [Faker::Games::DnD](doc/games/dnd.md)
   - [Faker::Games::Dota](doc/games/dota.md)
+  - [Faker::Games::DungeonsAndDragons](doc/games/dungeons_and_dragons.md)
```
```suggestion
```
looks like this line might be a leftover from the original PR

dollerbill commented:
@jameslvdb I didn't mean you had to do it right this second 😂
but your additions look great, this is probably as complete as a DND generator could possibly be!

stefannibrasil commented:
Hey there. In an effort to lighten our load as maintainers and be able to serve you better in the future, the faker-ruby team is working on cleaning out the cobwebs in this repo by pruning the backlog. As there are few of us, there are a lot of items that will simply never earn our attention in a reasonable time frame, and rather than giving you an empty promise, we think it makes more sense to focus on more recent issues. That means, unfortunately, that we must close this PR.

Don't take this the wrong way: our aim is not to diminish the effort people have made or dismiss problems that have been raised.

If you feel that we should reopen this PR, please do the following first:
- rebase with master
- verify this contribution is still relevant (i.e. hasn't been implemented/solved yet)
- run tests again

Then please let us know so that we can re-prioritize it. Thanks!
